### PR TITLE
#318: 'Add paths to icon with multipath' (closes #318)

### LINKED ISSUE #318: 'Add paths to icon with multipath'
See https://github.omnilab.our.buildo.io/buildo/labonline/issues/1522#issuecomment-4983

### PR DESCRIPTION
**Pull request by francescogior**
Issue #318

/cc @ascariandrea

### CODE REVIEW

File: src/Icon/Icon.js [hunk 1]
```diff
@@ -8,12 +9,14 @@ import cx from 'classnames';
   icon: t.maybe(t.Str),
   className: t.maybe(t.Str),
   style: t.maybe(t.Obj),
+  paths: t.maybe(t.refinements(t.Number, x => x - Math.floor(x) === 0 && x > 0)),
```
Comment (FrancescoCioria) [inline src/Icon/Icon.js]: please define `const Integer = t.refinement(t.Number, (n) => n % 1 === 0, 'Integer');` (copy-pasted from `tcomb` examples)
and then `paths: t.maybe(Integer)`

File: src/Icon/Icon.js [hunk 2]
```diff
@@ -8,12 +9,14 @@ import cx from 'classnames';
...
   onClick: t.maybe(t.Func)
 })
 export default class Icon extends React.Component {
 
   static defaultProps = {
     className: '',
```
Comment (FrancescoCioria) [inline src/Icon/Icon.js]: as you're already managing this component, please remove `style` and `className` from `defaultProps`

File: src/Icon/Icon.js [hunk 3]
```diff
@@ -24,9 +27,14 @@ export default class Icon extends React.Component {
     };
   }
 
-  template({ icon, className, style, onClick }) {
-    return icon ?
-      <i className={cx('icon', `icon-${icon}`, className)} style={style} onClick={onClick}/>
+  template({ icon, className, style, onClick, paths }) {
+    return icon ? (
+      <i className={cx('icon', `icon-${icon}`, className)} style={style} onClick={onClick}>
+        {paths === 1 ? null : range(paths).map(k => (
```
Comment (FrancescoCioria) [inline src/Icon/Icon.js]: replace with `paths > 1 &&` ?

Comment (francescogior) [inline src/Icon/Icon.js]: I grew up knowing that `React` doesn't like `false` as `children` but if you tell me that's ok I trust yoy

File: src/Icon/Icon.js (hunk 1 above)
Comment (francescogior) [inline src/Icon/Icon.js]: :+1:  but 
`const PositiveInteger = t.refinement(t.Number, (n) => n % 1 === 0 && n > 0, 'PositiveInteger');`

File: src/Icon/Icon.js (hunk 2 above)
Comment (francescogior) [inline src/Icon/Icon.js]: ok for `className` that is given to `cx` (`''` and `undefined` are treated the same), but what about style={null} ?

**Comment by ascariandrea** [comment]
Maybe create a new specific component for dealing with `icomoon` icons? 
Cause having a general `Icon` component that renders *n* span with `className={`path${x}`}` is too opinionated?

**Comment by francescogior** [comment]
added fixes, please @ascariandrea try this out

**Comment by francescogior** [comment]
you're right, but I think this was built already quite 'ad hoc' for `icomoon`

**Comment by FrancescoCioria** [comment]
> you're right, but I think this was built already quite 'ad hoc' for icomoon

yes, `Icon` currently works *only* for `icomoon`

**Comment by FrancescoCioria** [comment]
CI: `33:70  error  Infix operators must be spaced  space-infix-ops`

**Comment by ascariandrea** [comment]
Maybe we need a better name though..
I'm going to try it in the afternoon!

**Comment by ascariandrea** [comment]
Tested and it works well, for me it can be merged!

**Pull request merged**